crates/runtime/src/continuation.rs

```diff
@@ -203,11 +219,11 @@ pub fn resume(
             // calling trampoline to execute it.
 
             if cfg!(debug_assertions) {
-                let _parent = unsafe { (*contobj).parent };
+                let _parent_chain = unsafe { &(*contobj).parent_chain };
```

Comment:
```suggestion
                let _parent_chain = unsafe { contobj.as_ref().ok_or_else(|| TrapReason::user_with_backtrace(anyhow::anyhow!("attempt to dereference null continuation")))?.parent_chain };
```

Reply:
I don't really see the point in these null pointer checks (and the other, existing ones in this function): If `contobj` is a null pointer, the current version segfaults (which should be caught my the wasmtime signal handler and translated into a trap or panic). Your suggestion will lead to a trap telling you that we are trying to dereference a null pointer. In both cases, we will have to debug the situation, in which case the degree of information we obtain in both situations is almost the same, but your suggestion makes things more verbose and leads to more branch-y code.

Reply:
I think this is better initially. We can optimise it later. I think we should strive for safety first, performance second. Ultimately, when we are confident that the implementation is safe, we can start removing the redundant checks/branches.

Reply:
What I was saying was that these checks do not add any additional safety

Reply:
If that's the case then please add a `// SAFETY` arguing why this is in fact safe.

Reply:
Done